server/src/business-layer/services/StripeService.ts

```diff
@@ -452,6 +458,54 @@ export default class StripeService {
     }
   }
 
+  private async isLastSpotTaken(bookingSlotId: string): Promise<boolean> {
```

Comment:
also we can definitely write a test for this, if we move it to bookingslotservice

Reply:
I'd HIGHLY recommend doing this btw since we are going based off a code review only